### PR TITLE
Build CTS on windows or linux - Python script proposal

### LINKED ISSUE #537: Build CTS on windows or linux - Python script proposal
I am not sure if this is proper place to have such build script. The script is especially dedicated for windows but can build binaries on Linux as well.

### PR DESCRIPTION
**Pull request by gwawiork**
Fixes #537

### CODE REVIEW

File: build_win_lnx.py [hunk 1]
```diff
@@ -0,0 +1,176 @@
+import argparse
```
Comment (EwanC) [inline build_win_lnx.py]: `#!/usr/bin/python3`  ?

File: build_win_lnx.py [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+def clean_path(path):
+    if os.path.exists(path):
+        print("Cleaning path: ", path)
+        shutil.rmtree(path)
+    else:
+        print("The path: ", path, " does not exist")
+
+
+def process_command_line():
+    parser = argparse.ArgumentParser()
```
Comment (EwanC) [inline build_win_lnx.py]: a `description` argument here could he helpful

File: build_win_lnx.py [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+    parser.add_argument('--msvc_version', default="2017", choices=['2017', '2019'],
+                        help='choose msvc version')
+    parser.add_argument('--compiler', default="msvc", choices=['msvc', 'icc'],
+                        help='choose compiler')
+    parser.add_argument('--bitness', default="x64", choices=['x64', 'x86'],
+                        help='choose bitness')
+    parser.add_argument('--skip_clone', action='store_true',  help='skip cloning repositories')
+    parser.add_argument('--skip_icd', action='store_true',  help='skip building ICD project')
+    parser.add_argument('--skip_cts', action='store_true', help='skip building CTS project')
+    parser.add_argument('--build_target', default="ALL_BUILD", help='target to build (only Windows)')
```
Comment (EwanC) [inline build_win_lnx.py]: if this option only works on Windows can this `parser.add_argument` be wrapped in a `platform == 'win32':` so it users don't accidentally expect behaviour from it? Same for the `--msvc_version` and `--compiler` options.

File: build_win_lnx.py [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+        elif bitness == "x86" and msvc_version == "2019":
+            icd_cmd = icd_solution_cmd_win_x86_msvc_2019
+            if compiler == "msvc":
+                cts_cmd = cts_solution_cmd_win_x86_msvc_2019
+            elif compiler == "icc":
+                cts_cmd = cts_solution_cmd_win_x86_icc_2019
+
+    elif platform[:5] == "linux":
+        print("Linux operating system")
+        compiler = "gcc"
```
Comment (EwanC) [inline build_win_lnx.py]: Is this always true, or could the compiler be `clang` if an environment var like `CC/CXX` is set?

Comment (gwawiork) [reply build_win_lnx.py]: I've added only support for gcc. Feel free to extend support for any compiler you want

Comment (EwanC) [reply build_win_lnx.py]: Only supporting gcc is fine, was mostly concernced about the script printing `Compiler: gcc` when another compiler was actually being used by cmake. But now that message is guarded to windows only on line 117, so avoiding the potential for misleading output.

**Comment by lakshmih** [comment]
We would like to review this and will post comments shortly.

**Comment by lakshmih** [comment]
We would like to stick to cmake-based builds rather than a build script. We recommend moving any necessary logic into the CMake build and documentation to the readme (refer https://github.com/KhronosGroup/OpenCL-CTS/pull/1021)

**Comment by nikhiljnv** [comment]
Any reason we need separate scripts and can not reuse or build on top of https://github.com/KhronosGroup/OpenCL-CTS/blob/master/presubmit.sh ?

**Comment by gwawiork** [comment]
There are already scripts in repo

**Pull request closed without merging**